### PR TITLE
doc: update nfs doc for Kerberos setup of ganesha in Ceph

### PR DESCRIPTION
**Pull request by avanthakkar**
Update nfs doc for Kerberos setup of ganesha in Ceph , only for `FSAL_CEPH`

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/mgr/nfs.rst [hunk 1]
```diff
@@ -326,7 +326,8 @@ allowed to be any complete path hierarchy between ``/`` and the ``EXPORT {path}`
 
 .. note:: Specifying values for sectype that require Kerberos will only function on servers
           that are configured to support Kerberos. Setting up NFS-Ganesha to support Kerberos
-          is outside the scope of this document.
+          can be found in the `Kerberos setup for NFS Ganesha in Ceph <https://github.com/nfs-ganesha/nfs-ganesha/wiki/Kerberos-setup-for-NFS-Ganesha-in-Ceph>`_.
```
Comment (anthonyeleven) [inline doc/mgr/nfs.rst:329]: s/in the/here:/

File: doc/mgr/nfs.rst [hunk 2]
```diff
@@ -388,7 +389,7 @@ the supplied methods left-to-right.
...
```
Comment (anthonyeleven) [inline doc/mgr/nfs.rst:392]: s/in the/here:/

**Review by adk3798 — APPROVED** [review]

**Comment by phlogistonjohn** [comment]
I'm not a big fan of the documentation you linked to. It's very low level, it couples the system and container in ways I don't like, and it's lousy UX (IMO). But I guess I don't object to exposing people to the link if they need it.

I would much rather see ceph take an approach more similar to what Rook did for this topic.

**Comment by zdover23** [comment]
> I'm not a big fan of the documentation you linked to. It's very low level, it couples the system and container in ways I don't like, and it's lousy UX (IMO). But I guess I don't object to exposing people to the link if they need it.
> 
> I would much rather see ceph take an approach more similar to what Rook did for this topic.

Is this the Rook documentation that you'd like to emulate?

https://rook.io/docs/rook/v1.10/Storage-Configuration/NFS/nfs-advanced/?h=ganesha#adding-custom-nfs-ganesha-config-file-changes

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/59939 - Squid backport
https://github.com/ceph/ceph/pull/59940 - Reef backport
https://github.com/ceph/ceph/pull/59941 - Quincy backport

**Comment by phlogistonjohn** [comment]
No, I want the code to emulate the behavior of being able to set up tooling in containers, independent of the outer OS, like sssd to manage the kerberos and ldap properties automatically.

**Comment by phlogistonjohn** [comment]
Basically I'm a grouch who thinks this is something that should be done by automation, not people. ;-)

**Pull request merged**